korman/properties/modifiers/avatar.py

```diff
@@ -119,17 +134,62 @@ def harvest_actors(self):
     def logicwiz(self, bo, tree):
         nodes = tree.nodes
 
+        if not self.sitting_cam:
+            # Sitting Modifier
+            sittingmod = nodes.new("PlasmaSittingBehaviorNode")
+            sittingmod.approach = self.approach
+            sittingmod.name = "SittingBeh"
+
+            # Clickable
+            clickable = nodes.new("PlasmaClickableNode")
+            clickable.link_output(sittingmod, "satisfies", "condition")
+            clickable.clickable_object = self.clickable_object
+            clickable.bounds = find_modifier(self.clickable_object, "collision").bounds
+
+            # Avatar Region (optional)
+            region_phys = find_modifier(self.region_object, "collision")
+            if region_phys is not None:
+                region = nodes.new("PlasmaClickableRegionNode")
+                region.link_output(clickable, "satisfies", "region")
+                region.name = "ClickableAvRegion"
+                region.region_object = self.region_object
+                region.bounds = region_phys.bounds
+
+            # Facing Target (optional)
+            if self.facing_enabled:
+                facing = nodes.new("PlasmaFacingTargetNode")
+                facing.link_output(clickable, "satisfies", "facing")
+                facing.name = "FacingClickable"
+                facing.directional = True
+                facing.tolerance = self.facing_degrees
+            else:
+                # this socket must be explicitly disabled, otherwise it automatically generates a default
+                # facing target conditional for us. isn't that nice?
+                clickable.find_input_socket("facing").allow_simple = False
+        else:
+            sitting_pfm = sitting_pfms
+            sittingnode = self._create_python_file_node(tree, sitting_pfm["filename"], sitting_pfm["attribs"])
+            self._create_sitting_nodes(bo, tree.nodes, sittingnode)
```

Comment:
It looks like this section is basically a duplicate of the old code, and the old code received a minor adjustment to implement the new functionality. I think it would be more straightforward to handle the two cases together. I'll provide a sample suggestion below, please let me know if I misunderstood something.